draft-specifications/Addresses-References.md

```diff
@@ -0,0 +1,222 @@
+which together build up OCapN  (Object Capability Network) specifications.
+
+This specification uses the following other specifications:
+
+-   [Syrup](): The serialization format used to encode the in band
+    representation of the locator. 
+-   [Tagged Netstring](https://web.archive.org/web/20140210012056/http://tnetstrings.org/):
+    Specification for message framing.
+-   [OCapN Netlayers](): Specification to open a secure communication channel
+    between two sessions, often on different networks.
+-   [RFC3986: Uniform Resource Identifier (URI): Generic
+    Syntax](https://www.rfc-editor.org/rfc/rfc3986): For encoding URI
+    representations of locations.
+
+# [Abstract](#abstract)
+
+An OCapN **address** provides a node with all the information it needs to
+establish a session with another node.
+An address consists of the name of a network layer and transport-specific
+details.
+The name of the network layer captures multiple dimensions:
+
+- a **transport protocol** like TLS or Tor,
+- a **message framing protocol** like Netstring,
+- or, a combination of transport protocol and message framing protocol
+  like WebSocket over HTTP over TLS,
+- a **version** like CapTP Version 0
+
+An **reference** has an identifier and must have at least one address.
+
+- an **identifier** (string, typically a [Swiss
+  Number](http://wiki.erights.org/wiki/Swiss_number))
+- a list of **addresses** (strings)
+
+To send messages from one node to a reference in another node,
+the sender must establish a session with a protocol that both the sender
+and receiver support.
+
+> For example, if the sender node is a web page and the reference on a web
+> server, these nodes might both be able to connect using version 1 of
+> OCapN and rely on WebSocket for message framing and HTTP with TLS and a
+> Certificate Authority for transport.
+> The server might also provide addresses for that reference for TLS with a
+> Diffie-Helman key exchange, Netstring for message framing, and OCapN versions 1
+> and 2 for CapTP.
+
+# Network Layers
+
+OCapN specifies the names of multiple addressable network layers.
+A node may have multiple network layers and a reference may have multiple
+addresses and multiple addresses may share a network layer.
+
+## TCP with Syrup, Version 0
+
+For testing purposes only, the network layer named `tcp-syr0` combines:
+
+* TCP for sessions,
+* Netstring for message framing,
+* Syrup OCapN Version 0 (draft)
+
+> We advise the TCP transport to listen specifically on IP `127.0.0.1` (rather
+> than the typical default of `0.0.0.0`) to make the node unreachable from
+> other hosts and avoid use on multi-tenant hosts.
+> For private connections between a user and a user-agent on a multi-tenant
+> system, relay OCapN through a UNIX domain socket or a Windows named pipe.
+
+The Syrup serialization is a record with the symbol `'tcp-syr0` followed
+by a TCP address, including the port.
```

Comment:
Can we have port in a distinct field for JSON and syrup formats?
I'd rather parse a structure than run a sub-parser on a string.